File: npc-prompt/src/main/java/com/dleinbach/npcprompt/npc/Npc.java
package com.dleinbach.npcprompt.npc;

import com.dleinbach.npcprompt.ability.Ability;
import com.dleinbach.npcprompt.appearance.Appearance;
import com.dleinbach.npcprompt.bond.Bond;
import com.dleinbach.npcprompt.flaw.Flaw;
import com.dleinbach.npcprompt.ideal.Ideal;
import com.dleinbach.npcprompt.interaction.Interaction;
import com.dleinbach.npcprompt.mannerism.Mannerism;
import com.dleinbach.npcprompt.race.Race;
import com.dleinbach.npcprompt.talent.Talent;

public record Npc(Appearance appearance, Ability highAbility, Ability lowAbility, Talent talent,
                  Mannerism mannerisms, Interaction interaction, Ideal ideal, Bond bond, Flaw flaw, Race race) {
}

File: npc-prompt/src/main/js/npc/Npc.js
import NpcAttribute from './NpcAttribute';
import Race from './Race';

class Npc{

    constructor(appearance, highAbility, lowAbility, talent, mannerisms,
                    interaction, ideal, bond, flaw, race){
        this.talent = talent;
        this.appearance = appearance;
        this.highAbility = highAbility;
        this.lowAbility = lowAbility;
        this.mannerisms = mannerisms;
        this.interaction = interaction;
        this.ideal = ideal;
        this.bond = bond;
        this.flaw = flaw;
        this.race = race;
    }

    static parse(npc){
        return new Npc(NpcAttribute.parse(npc.appearance), new NpcAttribute(0, npc.highAbility), new NpcAttribute(0,npc.lowAbility),
            NpcAttribute.parse(npc.talent), NpcAttribute.parse(npc.mannerisms), NpcAttribute.parse(npc.interaction),
            NpcAttribute.parse(npc.ideal), NpcAttribute.parse(npc.bond), NpcAttribute.parse(npc.flaw), Race.parse(npc.race));
    }

    static empty(){
        return new Npc(new NpcAttribute(),new NpcAttribute(),new NpcAttribute(),new NpcAttribute(),
                       new NpcAttribute(),new NpcAttribute(),new NpcAttribute(),new NpcAttribute(),
                       new NpcAttribute(), new Race());
    }

}

export default Npc;